core/src/test/scala/unit/kafka/integration/BaseTopicMetadataTest.scala

```diff
@@ -133,6 +133,31 @@ abstract class BaseTopicMetadataTest extends ZooKeeperTestHarness {
   }
 
   @Test
+  def testAutoCreateTopicWithInvalidReplication {
+    val adHocProps = createBrokerConfig(2, zkConnect, interBrokerSecurityProtocol = Some(securityProtocol),
+      trustStoreFile = trustStoreFile)
+    // Set default replication higher than the number of live brokers
+    adHocProps.setProperty(KafkaConfig.DefaultReplicationFactorProp, "3")
+    // start adHoc brokers with replication factor too high
+    val adHocServer = createServer(new KafkaConfig(adHocProps))
+    // We are using the Scala clients and they don't support SSL. Once we move to the Java ones, we should use
```

Comment:
I think its because  `ClientUtils.fetchTopicMetadata` is used throughout a lot of tests and has not been migrated yet. 

This comment was copied verbatim from the setup section at the top of the test class: [BaseTopicMetadataTest.scala#L53-L55](https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/integration/BaseTopicMetadataTest.scala#L53-L55)

I wanted to make my "ad hoc" endpoint the same way the rest of the endpoints are made and copied to comment to ensure this line is not missed during any code migration.